.github/workflows/image.yml

```diff
@@ -8,154 +8,109 @@ on:
         with:
-          context: .
-          tags: ${{ steps.meta.outputs.tags }}
-          labels: ${{ steps.meta.outputs.labels }}
+          image_name: 'uptime-checker'
           platforms: linux/${{ matrix.platform }}
-          cache-from: type=gha,scope=${{ matrix.platform }}
-          cache-to: type=gha,mode=max,scope=${{ matrix.platform }}
-          annotations: ${{ steps.meta.outputs.annotations }}
-          provenance: mode=max
-          sbom: true
-          push: true
-
-  create-manifest:
-    name: Create multi-platform manifest
-    needs: build-main-or-release
-    if: |
-      github.repository_owner == 'getsentry'
-      && github.event_name == 'push'
-      && (github.ref == format('refs/heads/{0}', github.event.repository.default_branch) || startsWith(github.ref, 'refs/heads/release/'))
-
+          dockerfile_path: './Dockerfile'
+          build_args: UPTIME_CHECKER_GIT_REVISION=${{ github.event.pull_request.head.sha || github.sha }}
+          publish_on_pr: true  # required for self-hosted-end-to-end runs on pull requests
+          ghcr: true
+          google_ar: false
+          tag_suffix: -${{ matrix.platform }}
+          # note: nightly will only be tagged if on default branch
+          tag_nightly: true
+          tag_latest: false
+
+  assemble-ghcr:
+    needs: [build-multiplatform-image-ghcr]
+    if: ${{ github.repository_owner == 'getsentry' }}
+    runs-on: ubuntu-latest
     permissions:
       contents: read
       packages: write
-
-    runs-on: ubuntu-latest
     steps:
-      - uses: actions/checkout@f43a0e5ff2bd294095638e18286ca9a3d1956744 # v3
+      - name: Docker Login
+        run: docker login --username '${{ github.actor }}' --password-stdin ghcr.io <<< "$GHCR_TOKEN"
+        env:
+          GHCR_TOKEN: ${{ secrets.GITHUB_TOKEN }}
 
-      - name: Login to GHCR
-        uses: docker/login-action@74a5d142397b4f367a81961eba4e8cd7edddf772 # v3
-        with:
-          registry: ghcr.io
-          username: ${{ github.actor }}
-          password: ${{ secrets.GITHUB_TOKEN }}
+      - name: Set up Docker Buildx
+        uses: docker/setup-buildx-action@b5ca514318bd6ebac0fb2aedd5d36ec1b5c232a2 # v3.10.0
 
-      - name: Login to Docker Hub
-        uses: docker/login-action@74a5d142397b4f367a81961eba4e8cd7edddf772 # v3
-        with:
-          username: sentrybuilder
-          password: ${{ secrets.DOCKER_HUB_RW_TOKEN }}
+      - name: Create multiplatform manifests
+        run: |
+          docker buildx imagetools create \
+            --tag ghcr.io/getsentry/uptime-checker:${{ github.event.pull_request.head.sha || github.sha }} \
+            ghcr.io/getsentry/uptime-checker:${{ github.event.pull_request.head.sha || github.sha }}-amd64 \
+            ghcr.io/getsentry/uptime-checker:${{ github.event.pull_request.head.sha || github.sha }}-arm64
 
-      - name: Create and push multi-platform manifest
+      - name: Tag nightly
```

Comment:
We shouldn't need an extra step here, can just directly add another `--tag` above